specification/src/chapter3.adoc

```diff
@@ -1,45 +1,312 @@
+|===
+
+==== sPMP
+
+https://github.com/riscv/riscv-spmp
+
+_Supervisor PMP (sPMP)_ enables supervisor mode to control physical memory access for U mode.
+
+sPMP allows supervisor mode to restrict its own access to memory allocated to lower privilege levels. This can be used to mitigate against privilege escalation attacks, for example.
+
+When combined with H-extension, sPMP can be nested so that the hypervisor can control memory allocations to its guests, and each guest can control its own memory allocations to its workloads.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| Either sPMP, 1st-stage or G-stage page tables MUST be used to protect Supervisor domain H/S-mode from lower privilege levels.
+|===
+
+==== MMU
+
+https://github.com/riscv/riscv-isa-manual/releases/tag/Priv-v1.12[Privileged ISA]
+
+_Memory management unit (MMU)_ enables address translation and protection for:
+
+* Isolating an OS from workloads on a system without H-extension (one-stage translation)
+* Isolating a hypervisor from a guest, on a system with H-extension (two-stage translation)
+ 
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| Either PMP/ePMP or MTT MUST be used to protect M-mode from lower privilege levels.
+|===
+
+==== Supervisor domains
+
+https://github.com/riscv/riscv-smmtt
+
+Supervisor domains allow software components on the same hart to be developed, certified, deployed and attested independently of each other.
+
+A supervisor domain is a compartment above M-mode, physically isolated - memory, execution state, and devices - from other supervisor domains regardless of privilege level (below M-mode). Isolation and context switching between supervisor domains are managed by M-mode firmware. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NN
+| Either PMP/ePMP or MTT MUST be used to enforce physical memory isolation boundaries for supervisor domains, and to protect machine mode from any supervisor domain.
+
+|===
+
+PMP can be used for more static and deterministic use cases.
+
+MTT can be used where more fine grained dynamic resource management across supervisor domain boundaries is required.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+
+| ID#     
+| Requirement
+
+| CAT_NNN
+| A system supporting supervisor domains MUST support supervisor domain extensions for interrupts (Smsdia) and performance counters (TBD), and SHOULD support supervisor domain extensions for external debug (Smsdsd TBD).
+
+|===
+
+Interrupts: https://github.com/riscv/riscv-smmtt +
+External debug: https://github.com/riscv-non-isa/riscv-external-debug-security +
+Performance counters: 
+
+These extensions enable management of interrupts, external debug, and performance counters across supervisor domain boundaries. M-mode firmware should context switch hart HPM event/counters to manage isolation of performance counters:
+
```

Comment:
Since they are context switched, there is no ISA extension required for performance counters.